### PR TITLE
feat: generate preview versions

### PR DESCRIPTION
**Pull request by nguyentoanit**
Demo: https://github.com/ScaleLeap/selling-partner-api-sdk/pull/148

### CODE REVIEW

**Comment by moltar** [comment]
Task linked: [GHORG-67 Generate all api versions](https://app.clickup.com/t/2250003/GHORG-67)

**Comment by nguyentoanit** [comment]
@moltar: 
Can you setup a host to contain docs?

**Comment by moltar** [comment]
> @moltar:
> Can you setup a host to contain docs?

1. Domain provisioned and linked to Netlify: `selling-partner-api-sdk.scaleleap.org`
2. Added secrets to GH repo under: `NETLIFY_AUTH_TOKEN` and `NETLIFY_SITE_ID`

**Comment by nguyentoanit** [comment]
Sample structure for preview version: 
- API model folder name: `catalog-items-api-model-v20201201`
- API client folder name: `catalog-items-api-client-v20201201`
- API client class name: `CatalogItemsApiClientV20201201`
- Exports: `ItemImageVariantEnum as CatalogItemsApiModelV20201201ItemImageVariantEnum`

File: utils/generate.ts [hunk 1]
```diff
@@ -1,46 +1,43 @@
...
-        outputPath,
-      }
+      const [defaultVersion, ...previewVersions] = fs.readdirSync(path.resolve(rootPath, dirname))
 
-      log.info(`Starting generating ${dirname}`)
-      // TODO: throw an error when command occurs error
-      execSync(command)
-      log.info(`Finished generating ${dirname}`)
+      const stableVersionAPIModel = generateModelForStableVersion(rootPath, dirname, defaultVersion)
+      // eslint-disable-next-line no-await-in-loop
```
Comment (moltar) [inline utils/generate.ts:30]: I think there is a reason it is complaining :)

Maybe try `for await (....)` loop.

File: utils/generator/api-model-generator.ts [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 import log from 'fancy-log'
 import { camelCase, upperFirst } from 'lodash'
+import path from 'path'
 import { Project } from 'ts-morph'
 
 import { APIModel } from './api-model'
 import { API_MODEL_FILE_NAME, TS_CONFIG_FILE_PATH, TS_LIB_FOLDER_PATH } from './constants'
+import { Parser } from './parser'
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const fs = require('fs')
```
Comment (moltar) [inline utils/generator/api-model-generator.ts:11]: Why is this a `require`?

**Review by moltar — CHANGES_REQUESTED** [review]

File: utils/generate.ts (hunk 1 above)
Comment (nguyentoanit) [reply utils/generate.ts]: Resolved: https://github.com/ScaleLeap/selling-partner-api-sdk/pull/147/commits/0b7bc1e6257869ebd82b3d8b4d68835d8b4fbd8e#diff-ba48db8f257b58c9b15436ddd7e5a402ae823cc9b5d5dbfa237903d8ed037b0bR39

**Review by moltar — APPROVED** [review]

**Pull request merged**